src/minimalistic-area-card.ts

```diff
@@ -521,19 +523,38 @@ export class MinimalisticAreaCard extends LitElement implements LovelaceCard {
     return !!stateObj.attributes.unit_of_measurement || !!stateObj.attributes.state_class;
   }
 
-  private computeStateValue(stateObj: HassEntity, entity?: EntityRegistryDisplayEntry) {
-    if (this.isNumericState(stateObj)) {
+  private computeStateValue(
+    stateObj: HassEntity,
+    entityConf: ExtendedEntityConfig,
+    entity?: EntityRegistryDisplayEntry,
+  ) {
+    if (!stateObj.attributes) {
+      stateObj.attributes = {};
+    }
+    const units = this._getOrDefault(
+      entity?.entity_id,
+      entityConf.unit_of_measurement,
+      this._getOrDefault(entity?.entity_id, stateObj.attributes.unit_of_measurement, ''),
+    );
+    const _fmt = function (str: string, units: string): string {
+      return `${str}${units ? ' ' + units : ''}`;
+    };
+
+    if (['unavailable', 'unknown', 'idle'].includes(String(stateObj.state).toLowerCase())) {
+      return null;
+    } else if (['off', 'on', 'true', 'false'].includes(String(stateObj.state).toLowerCase())) {
+      return units;
+    } else if (this.isNumericState(stateObj)) {
```

Comment:
I think all the "elses" here don't need to be elses, since the clause above returns.

It's more a stylistic choice, but the code could tighten up if you wanted.